EventFilter/SiPixelRawToDigi/plugins/SiPixelRawToDigi.cc

```diff
@@ -241,30 +245,58 @@ void SiPixelRawToDigi::produce( edm::Event& ev,
 	  // in the configurable error list in the job option cfi.
 	  // Code needs to be here, because there can be a set of errors for each 
 	  // entry in the for loop over PixelDataFormatter::Errors
-	  if(!tkerrorlist.empty() || !usererrorlist.empty()){
-	    DetId errorDetId(errordetid);
-	    edm::DetSet<SiPixelRawDataError>::const_iterator itPixelError=errorDetSet.begin();
-            for(; itPixelError!=errorDetSet.end(); ++itPixelError){
-              // fill list of detIds to be turned off by tracking
-              if(!tkerrorlist.empty()) {
-	        std::vector<int>::iterator it_find = find(tkerrorlist.begin(), tkerrorlist.end(), itPixelError->getType());
-	        if(it_find != tkerrorlist.end()){
+	  edm::DetSet<SiPixelRawDataError>::const_iterator itPixelError=errorDetSet.begin();
+	  std::vector<PixelFEDChannel> disabledChannelsDetSet;
+
+	  for(; itPixelError!=errorDetSet.end(); ++itPixelError){
+	    // For the time being, we extend the error handling functionality with ErrorType 25
+	    // In the future, we should sort out how the usage of tkerrorlist can be generalized
+	    if (itPixelError->getType()==25) {
+	      assert(itPixelError->getFedId()==fedId);
+	      const sipixelobjects::PixelFEDCabling* fed = cabling_->fed(fedId);
+	      if (fed) {
+		cms_uint32_t linkId = (itPixelError->getWord32() >> formatter.LINK_shift) & formatter.LINK_mask;
```

Comment:
this [starting from ```if (itPixelError->getType()==25```] looks like a hack for a work that is designed to be done by the formatter.

Reply:
Unfortunately, this is the right place to populate the disabled_channelcollection staying consistent with the earlier code (and avoid a potential double-counting of FED 25 errrors)

Reply:
it's fine to populate the disable collection here.
The point of the comment was that SiPixelRawToDigi should not do the word decoding and it instead is a functionality to be done by the formatter.